Possible compiler bug in handling of comptime conditionals

### Zig Version

0.12.0

### Steps to Reproduce and Observed Behavior

When compiling this code
````
pub fn bind_text(comptime T: type, stmt: *s.sqlite3_stmt, column: c_int, value: T) void {
    if (T == []const u8) {
        const buffer_size: u8 = 128;
        const buffer: [buffer_size:0]u8 = undefined;
    }
    var result_code: c_int = s.sqlite3_bind_text(
        stmt,
        column,
        if (T == [:0]const u8 or T == [:0]u8)
            &value[0]
        else if ((T == [*:0]const u8) or (T == *const u8) or (T == [*c]const u8))
            value
        else if (T == []const u8) {
            blk: {
                if (value.len >= buffer_size)
                    panic("bind_text: buffer too small to hold null-terminated value\n", .{});
                @memcopy(buffer[0..value.len], value);
                buffer[value.len] = null_char;
                break :blk &buffer[0];
            }
        } else panic("Bind-text called with invalid type: {s}", .{@typeName(@TypeOf(value))}),
        -1,
        s.SQLITE_STATIC,
    );
    if (result_code != s.SQLITE_OK) {
        panic("bind_text failed. Result code was {d}, column was {d}", .{ result_code, column });
    }
}
````
I get the following errors:
````
zig build-exe  --verbose-link --main-pkg-path .. -fsingle-threaded -I/usr/include  -lc -lsqlite3 vanguard_importer.zig
/usr/home/dca/Software/newcash_c/zig_library/library.zig:180:15: error: unused local constant
        const buffer: [buffer_size:0]u8 = undefined;
              ^~~~~~
/usr/home/dca/Software/newcash_c/zig_library/library.zig:191:34: error: use of undeclared identifier 'buffer_size'
                if (value.len >= buffer_size)
                                 ^~~~~~~~~~~
````
The first line of code is line 177.

This occurred on a FreeBSD 13.2 system. I built zig from the source code on 9/3.

### Expected Behavior

I think this function should compile without error. The 'if' at line 178 should cause the definitions
of buffer and buffer_size to be emitted and the 'if' at line 189 should cause the block labeled 'blk' to be emitted, that reference 'buffer' and 'buffer_size'.

Perhaps I've missed something, but I don't understand why these errors are occurring.

--- a/doc/langref.html.in
+++ b/doc/langref.html.in
@@ -7015,7 +7015,9 @@ test "try to compare bools" {
       </p>
       {#code_begin|syntax|compiler_generated_function#}
 fn max(a: bool, b: bool) bool {
-    return a or b;
+    {
+        return a or b;
+    }
 }
       {#code_end#}
       <p>

